src/include/btree_cmp_inline.h

```diff
@@ -245,15 +246,17 @@ __wt_lex_compare_skip(const WT_ITEM *user_item, const WT_ITEM *tree_item, size_t
     /* Contents are equal up to the smallest length. */
     if (ret_val == 0)
         ret_val = ((usz == tsz) ? 0 : (usz < tsz) ? -1 : 1);
-        /* In diagnostic mode ensure that short comparisons work as expected */
-#ifdef HAVE_DIAGNOSTIC
```

Comment:
Adding an additional comparison into every comparison might be noticeable for fast workloads.  I'm not sure if this is worth that risk (and I don't think it's important enough to try it out in the field to find out). I'd be inclined to keep this diagnostic only for now, I think.

Reply:
I've realized that my comment here was a bit hard to understand. What I was saying is that an `ifdef` is compiled in or out, and doesn't incur any runtime overhead if it is compiled out. Wheras an `if (...)` always incurs a runtime overhead - generally that overhead is small enough that we don't worry about it, but in this code - which is the core comparison function for the btree we care about performance enough to avoid the comparison in production builds.

So I'd prefer for this to remain within an `ifdef`, as well as being optionally available via a configuration option.